### PR TITLE
Fix lingering tiles during world teleports

### PR DESCRIPTION
**Pull request by alejandroliu**
New Pull Request in switchLevel

When somebody teleports between worlds Tile objects that used to be in the departure world keep lingering in the destination world.  These can be seen but cannot be interact with.
This pull request introduces a work around to solve these.

This is how the issue looks like:

### CODE REVIEW

**Comment by matcracker** [comment]
:+1:

**Comment by alejandroliu** [comment]
@shoghicp , is there anything else you need before this Pull Request can be merged?

**Comment by AndreyNazarchuk** [comment]
Yay I was wondering ow to fix this :) :+1:

File: src/pocketmine/Player.php [hunk 1]
```diff
@@ -590,6 +590,17 @@ public function isSleeping(){
 	protected function switchLevel(Level $targetLevel){
 		$oldLevel = $this->level;
 		if(parent::switchLevel($targetLevel)){
+			//TODO HACK: removes tile entities that linger whenever you teleport
+			// to a different world
+			$pk = new UpdateBlockPacket();
+			foreach($oldLevel->getTiles() as $tile){
+				$pk->records[] = [$tile->x, $tile->z, $tile->y, 0, 0, UpdateBlockPacket::FLAG_NONE];
+			}
+			if(count($pk->records)){
```
Comment (iksaku) [inline src/pocketmine/Player.php]: ``count($k->records) > 0``

Comment (alejandroliu) [inline src/pocketmine/Player.php]: It is enough to check that it is a non-zero value.

**Comment by shoghicp** [comment]
@alejandroliu $oldLevel->getTiles() will send more data than needed, as that will get all from the entire world. What you want is to do that on unloadChunk() (so it's not implemented in two places) so you can get the Chunk tiles directly.

**Comment by alejandroliu** [comment]
Will take a look at unloadChunk.

From what examining the code, $oldLevel()->getTiles() only sends the tiles that are loaded in memory.  Since chunks are dynamically loaded/unloaded, it should only send active tiles (which shouldn't be that many).

**Comment by alejandroliu** [comment]
Oh, yes, the unloadChunk bit is much better... I will be updating this pull request.

**Comment by shoghicp** [comment]
Can you squash this into a single commit so we don't have all the tries? Then force-push into your branch.

**Comment by alejandroliu** [comment]
Created a new PR #3197 .  This one is a single commit.

**Pull request closed without merging**